dubbo-metadata-report-extensions/dubbo-metadata-report-etcd/src/main/java/org/apache/dubbo/metadata/store/etcd/EtcdMetadataReport.java

```diff
@@ -126,6 +126,16 @@ public String getServiceDefinition(MetadataIdentifier metadataIdentifier) {
         return etcdClient.getKVValue(getNodeKey(metadataIdentifier));
     }
 
+    @Override
+    public boolean registerServiceAppMapping(String key, String group, String content, Object ticket) {
+        String appKey = toRootDir() + "mapping-data" + PATH_SEPARATOR + key + PATH_SEPARATOR + group;
+        if (!etcdClient.put(appKey, content)) {
```

Comment:
Does etcd support cas?

Reply:
> Does etcd support cas?
yes

Reply:
1，The group should be in the middle？
     like this： toRootDir() + group + PATH_SEPARATOR + serviceKey
2，etcd's put method does not support cas.

Reply:
1，The group is not at the end？
like this： toRootDir() + group + PATH_SEPARATOR + serviceKey,
2，etcd's put method does not support cas.

Reply:
> 1，The group is not at the end？ like this： toRootDir() + group + PATH_SEPARATOR + serviceKey, 2，etcd's put method does not support cas.

1、Can adjust
2、Does not support cas，etcd3 transaction?

Reply:
etcd3 transactions support cas